portfolio/src/main/webapp/script.js

```diff
@@ -80,25 +66,64 @@ yogaReveal.onclick = function() {
 }
 
 
-function getName() {
-    console.log('Fetching Name');
+// To parse JSON into a usable object for javascript
+function getComments() {
+
+    // fetch data from server, parse as json file, then reference coms as an object
+    fetch('/data').then(response => response.json()).then((comments) => {
+        const commentList = document.getElementById("comment-container");
+        console.log("fetching comments");
+        comments.forEach((comments) => {
+            commentList.appendChild(createListElement(comments));
+        })
+    });
+}
+
+// For learning promises
+function getCommentsLong() {
+    console.log('fetching JSON');
 
-    // request html from servlet data
     const responsePromise = fetch('/data');
 
     responsePromise.then(handleResponse);
 }
 
 function handleResponse(response) {
-    console.log('Handle name response');
-
-    // converts response into text
-    const textPromise = response.text();
+    console.log('convert JSON promise stream to text');
+    const textPromise = response.json();
+    console.log(textPromise);
+    textPromise.then(addListToDom);
+}
 
-    textPromise.then(addName);
+function addListToDom(list) {
+    const commentList = document.getElementById("comment-container");
+    commentList.innerText = '';
+    // list.pageOne does not work as a JSON element should, should this work with array format?
+    commentList.appendChild(
+      createListElement("Page One: " + list[0]));
+    commentList.appendChild(
+      createListElement("Page Two: " + list[1]));
+    commentList.appendChild(
+      createListElement("Page Three: " + list[2]));
 }
```

Comment:
Great job exploring the different uses of promises.  Currently, this and the getCommentsAsyncAwait function aren't used.  That is fine but I would suggest moving createListElement above getCommentsLong to separate working and non-active code.

 Maybe also add a comment above getCommentsLong that says something like:

`
// The following code was used for learning promises. It is currently not being used and can be removed in the future once it isn't needed.
function getCommentsLong() {
`